cmds/exp/dmidecode/testdata/dmidecode_diff.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```

Comment:
Why is there a bash dependency in here? This can't stay.

Reply:
I'd prefer a test that uses QEMU.

The current testsetup runs dmidecode on whatever machine you happen to be on and diffs the output. I think that's useful as a script or something, but a real regression test should be reproducible.

take a look at the tests we do e.g. `pkg/gpio`:
- set up a VM: https://github.com/u-root/u-root/blob/master/pkg/gpio/gpio_integration_test.go
- test that runs inside VM: https://github.com/u-root/u-root/blob/master/pkg/gpio/gpio_linux_test.go

we can control smbios tables in qemu easily with command-line arguments.

the qemu.Options struct also takes "devices" which you can use to pass those command-line arguments:

```go
func TestIntegration(t *testing.T) {
	vmtest.GolangTest(t, []string{"github.com/u-root/u-root/pkg/gpio"}, &vmtest.Options{
		QEMUOpts: qemu.Options{
			// Make GPIOs nums 10 to 20 available through the
			// mockup driver.
			KernelArgs: "gpio-mockup.gpio_mockup_ranges=10,20",
                        Devices: []qemu.Device{
                                qemu.ArbitraryArgs{"-smbios", "type=0,..."},
                        },
		},
	})
}
```

Reply:
Oh wow, nevermind me, I see the static files. Yeesh.

I'd still like having a QEMU test like this.

Reply:
right, so this script is not currently used in tests, but can be used manually.
tests diff our own output against the checked-in golden (the `.txt` file), the script diffs our golden output against dmidecode(8).
i don't think we should be beholden to existing dmidecode and its output forever, these diffs are something to be aware of but the build shouldn't necessarily break if upstream output changes, for example.